src/test/java/net/sf/jabref/importer/fileformat/RepecNepImporterTest.java

```diff
@@ -0,0 +1,122 @@
+import org.junit.Before;
+import org.junit.Test;
+
+import net.sf.jabref.Globals;
+import net.sf.jabref.JabRefPreferences;
+import net.sf.jabref.bibtex.BibtexEntryAssert;
+import net.sf.jabref.importer.OutputPrinterToNull;
+import net.sf.jabref.model.entry.BibEntry;
+
+public class RepecNepImporterTest {
+
+    @Before
+    public void setUp() throws Exception {
+        if (Globals.prefs == null) {
+            Globals.prefs = JabRefPreferences.getInstance();
+        }
+    }
+
+    @Test
+    public final void testIsRecognizedFormat() throws IOException {
+        RepecNepImporter testImporter = new RepecNepImporter();
+
+        //test with non recognized format files
+        List<String> notAccepted = Arrays.asList("RepecNep1.xml", "CopacImporterTest1.txt", "RisImporterTest1.ris",
+                "CopacImporterTest2.txt", "IEEEImport1.txt");
+        for (String s : notAccepted) {
+            try (InputStream stream = RepecNepImporter.class.getResourceAsStream(s)) {
+                Assert.assertFalse(testImporter.isRecognizedFormat(stream));
+            }
+        }
+
+        //test with recognized format files
+        List<String> accepted = Arrays.asList("RepecNepImporterTest1.txt", "RepecNepImporterTest2.txt",
+                "RepecNepImporterTest3.txt");
+        for (String s : accepted) {
+            try (InputStream stream = RepecNepImporter.class.getResourceAsStream(s)) {
+                Assert.assertTrue(testImporter.isRecognizedFormat(stream));
+            }
+        }
+    }
+
+    @Test
+    public final void testImportEntries() throws IOException {
+        RepecNepImporter testImporter = new RepecNepImporter();
+
+        List<BibEntry> entries = testImporter.importEntries(
+                RepecNepImporter.class.getResourceAsStream("RepecNepImporterTest1.txt"), new OutputPrinterToNull());
+        Assert.assertEquals(2, entries.size());
+
+        entries = testImporter.importEntries(RepecNepImporter.class.getResourceAsStream("RepecNepImporterTest3.txt"),
+                new OutputPrinterToNull());
+        Assert.assertEquals(17, entries.size());
+
+        entries = testImporter.importEntries(RepecNepImporter.class.getResourceAsStream("RepecNepImporterTest4.txt"),
+                new OutputPrinterToNull());
+        Assert.assertEquals(29, entries.size());
+
+        try {
+            entries = testImporter.importEntries(null, new OutputPrinterToNull());
+        } catch (Exception e) {
+            //catch Exception
+        }
+
+    }
+
+    @Test
+    public final void testImportSingleEntries() throws Exception {
+        RepecNepImporter testImporter = new RepecNepImporter();
+
+        //check all 4 test files and take the first two entries and compare with the bib file
+        for (int i = 1; i < 5; i++) {
+            try (InputStream in = RepecNepImporter.class.getResourceAsStream("RepecNepImporterTest" + i + ".txt");
+                    InputStream bibIn1 = RepecNepImporter.class
```

Comment:
The problem with tests like this is that they can break because of three reasons:
1. The actual importer has a bug
2. The comparison logic has a bug
3. The bib-parser has a bug
Using direct assertEquals on the resulting bibentries (almost) eliminates possibility 2+3.
I think the main problem is that the entries are so big. In an idea world, you have many tests with small entries. Then if a tests fails you know exactly which feature has a bug.

On the other hand, I can see why it might be good idea to compare with the bibfile. And since the importer are relatively stable, I think it is ok to leave the tests as they are right now.

Ps: Something which worries me a little bit: while adding all these tests for the importer almost no bug where found. I find this strange.